src/components/galery-modal/index.js

```diff
@@ -1,19 +1,70 @@
 import { ModalContainer } from './styled'
 import { useState } from 'react'
 import closeIcon from './close-icon.svg'
+import { Swiper, SwiperSlide } from 'swiper/react'
+import { FreeMode, Navigation, Thumbs } from 'swiper'
+import nossacasa from './nossacasa.png'
+import casa from './casa.png'
+import casadentro from './casadentro.png'
+
+import 'swiper/css'
+import 'swiper/css/free-mode'
+import 'swiper/css/navigation'
+import 'swiper/css/thumbs'
+
+const IMAGES = [nossacasa, casa, casadentro, nossacasa, nossacasa, nossacasa]
 
 const ModalGallery = () => {
   const [toggleGallery, setToggleGallery] = useState(false)
+  const [thumbsSwiper, setThumbsSwiper] = useState(null)
 
   const toggleGalleryHandler = () => {
     setToggleGallery(!toggleGallery)
   }
-
+  const closeAndRefresh = () => {
+    toggleGalleryHandler()
+    window.location.reload()
```

Comment:
Eu estou na dúvida se esta seria a melhor solução para corrigir o bug da modal, acho forçar uma re-renderização do DOM inteiro vai contra o design do React ou de Single Page Applications de uma forma geral.

Dito isso, eu ainda não tenho uma alternativa melhor para propor, então para não bloquear o desenvolvimento do card, minha sugestão seria seguir com esta implementação neste momento (assumindo que isso não quebra outras coisas na aplicação) e criar um débito técnico no backlog para que possamos investigar e encontrar uma solução mais adequada em algum momento mais oportuno.